packages/jetbrains-plugin/src/test/kotlin/com/mongodb/jbplugin/fixtures/UiTestExtensions.kt

```diff
@@ -126,18 +129,26 @@ private class UiTestExtension :
         remoteRobot.keyboard { escape() }
         remoteRobot.closeProject()
 
-        requiresProject?.let {
+        requiresProject?.let { project ->
             // If we have the @RequireProject annotation, load that project on startup
             remoteRobot.openProject(
-                Path("src/test/resources/project-fixtures/${it.value}").toAbsolutePath().toString(),
+                Path(
+                    "src/test/resources/project-fixtures/${project.value}"
+                ).toAbsolutePath().toString(),
             )
 
-            if (it.smartMode) {
+            if (project.smartMode) {
                 remoteRobot.ideaFrame().disablePowerSaveMode()
+                remoteRobot.openGradleToolWindow().also {
+                    it.ensureGradleProjectsAreSynced()
+                }
+                // This ideally should not be needed anymore, but we still perform this
+                // to wait for smart mode to kick-in before doing anything else.
                 remoteRobot.ideaFrame().waitUntilProjectIsInSync()
```

Comment:
I think it's fine to keep this. If the project is already in sync and in smart mode, it should be a noop, so I don't think it will slow the test down much.